File: PortsAndAdapters/PortsAndAdaptersHelloWorld - After/Gui/Window.cs
using System.Security.Cryptography;
using DataAccess.Ports;

namespace Gui
{
    public class Window
    {
      readonly IDomainLogic _logic;

      public Window(IDomainLogic domainLogic)
      {
        _logic = domainLogic;
      }

      public void OnSubmitClicked()
      {
        _logic.HandleAddEmployeeRequest();
      }
    }
}

File: Java/PortsAdapters/ports-and-adapters-after/src/main/java/com/example/gui/Window.java
package com.example.gui;

import com.example.ports.views.Application;

public class Window {
  private final Application domainLogic;

  public Window(final Application domainLogic) {
    this.domainLogic = domainLogic;
  }

  public void onSubmitClicked()
  {
    domainLogic.handleAddEmployeeRequest();
  }
}
